scripts/gen_mmu.py

```
@@ -0,0 +1,425 @@
+                                               size = mem_size,
+                                               pde_index = pde_index,
+                                               pte_valid_addr_start = pte_valid_addr_start,
+                                               pte_valid_addr_end = pte_valid_addr_end,
+                                               permissions = basic_mem_region_values[2])
+
+            # print(mem_region_values)
+        # print(hex(mem_region_values.start_addr),hex(mem_region_values.pde_index),
+        #       hex(mem_region_values.size))
+        if(mem_region_values.pde_index in list_of_pde.keys()):
+            validation_issue_memory_overlap = True
+        page_tables_list.append(mem_region_values.pde_index)
+        list_of_pde[pde_index] = mem_region_values
+
+        #add necessary pde to get the size correct for the mem region
+        overflow_size = basic_mem_region_values[1] - \
+                        ((pte_valid_addr_end - pte_valid_addr_start) * 4096)
+        # print("basic_mem_region_values[1]",hex(basic_mem_region_values[1]),
+        #       hex(overflow_size),list_of_pde )
+
+        for extra_pde in range(pde_index+1, get_pde_number(
+                mem_region_values.start_addr + basic_mem_region_values[1])+1):
+
+            extra_pde_start_address = extra_pde*(4096*1024)
+
+            extra_pte_valid_addr_end = get_pte_number(mem_region_values.start_addr
+                                                      + overflow_size)
+            # if extra_pte_valid_addr_end >= 1023:
+            #     extra_pte_valid_addr_end =  1023
+
+            extra_region_size = overflow_size
+            if overflow_size > (1024*4096):
+                extra_region_size = 1024*4096
+                extra_pte_valid_addr_end =  1024
+
+            if(extra_pde in list_of_pde.keys()):
+                validation_issue_memory_overlap = True
```

should throw an exception here